src/ts/commands.ts

```diff
@@ -131,32 +131,20 @@ export const selectMatchCommand = <TMatch extends IMatch>(
 };
 
 /**
- * Set the debug state. Enabling debug mode provides additional marks
- * to reveal dirty ranges and ranges sent for matching.
+ * Set a configuration value.
  */
-export const setDebugStateCommand = (debug: boolean): Command => (
-  state,
-  dispatch
-) => {
-  if (dispatch) {
-    dispatch(
-      state.tr.setMeta(PROSEMIRROR_TYPERIGHTER_ACTION, setDebugState(debug))
-    );
-  }
-  return true;
-};
-
-/**
- * When enabled, the plugin will queue match requests as soon as the document is modified.
- */
-export const setRequestOnDocModifiedState = (
-  requestMatchesOnDocModified: boolean
+export const setConfigValueCommand = <
+  ConfigKey extends keyof IPluginConfig,
+  ConfigValue extends IPluginConfig[ConfigKey]
+>(
```

Comment:
I'm a big fan of this Typescript pattern – it ensures that
- the configuration keys are typesafe
- the configuration values are typesafe for that key